plutus-core/generators/Language/PlutusCore/Generators/NEAT/Spec.hs

```diff
@@ -166,13 +177,27 @@ prop_agree_Ck_Cek tyG tmG = do
   tm <- withExceptT GenError $ convertClosedTerm tynames names tyG tmG
   withExceptT TypeError $ checkType tcConfig () tm (Normalized ty)
 
-  -- check if CK and CEK give the same output
   tmCek <- withExceptT CekP $ liftEither $
     evaluateCek testBuiltinsRuntime tm `catchError` handleError ty
 
   tmCk <- withExceptT CkP $ liftEither $
     evaluateCk testBuiltinsRuntime tm `catchError` handleError ty
-  unless (tmCk == tmCek) $ throwCtrex (CtrexTermEvaluationMismatch tyG tmG [tmCek,tmCk])
+
+  -- check if CK and CEK give the same output
+  unless (tmCk == tmCek) $
+    throwCtrex (CtrexTermEvaluationMismatch tyG tmG [tmCek,tmCk])
+
+  -- erase CEK output
+  let tmCekU = U.erase tmCek
+
+  -- run untyped CEK on erased input
+  tmUCek <- withExceptT UCekP $ liftEither $
+    U.evaluateCek testBuiltinsRuntime (U.erase tm) `catchError` handleUError
```

Comment:
``blah `catchError` handleUError`` is called for all the machines. Could we drop it then and still expect the results to be equal?

Reply:
I don't want a user error to be considered a counter example/test failure found be NEAT. This handler catches user errors and turns them the error terms which are returned 'successfully'.

Reply:
Ah, sure, silly me.